client/verta/verta/runtime.py

```diff
@@ -0,0 +1,240 @@
+    """
+    Return the current thread-local variable for validate or initialize a
+    new boolean.
+    """
+    if not hasattr(_THREAD, 'validate'):
+        _set_validate_flag(False)
+    return _THREAD.validate
+
+
+def _set_validate_flag(flag: bool) -> bool:
+    """
+    Sets the thread-local validate flag boolean value.
+    """
+    _THREAD.validate = flag
+    return _THREAD.validate
+
+
+def _delete_validate_flag() -> None:
+    """
+    Drop the `validate` attribute from the thread local variable.
+    """
+    _THREAD.__delattr__('validate')
+
+
+def _validate_json(value: Any) -> str:
+    """
+    Attempt to serialize the provided value to JSON.
+    Raise an error on failure.
+    """
+    try:
+        return json.dumps(value)
+    except TypeError:
+        raise
+
+
+def _validate_s3(
+        value: str,
+        pattern: re.Pattern = _S3_REGEX
+    ) -> None:
+    """
+    Check the provided value for any special characters that would
+    violate Amazon S3 object key naming rules. Raise ValueError if any
+    are found.
+    """
+    if len(value) > 100:
+        raise ValueError(" provided key value must be 100 characters or less in length.")
+    if not pattern.match(value):
+        raise ValueError(f" provided value \"{value}\" contains non-alphanumeric "
+                         f"characters. (dashes and underscores permitted)")
+
+
+def log(key: str, value: Any) -> None:
+    """
+    Updates current logging dict with provided key and value.
+    For use within the scope of a model's :meth:`~verta.registry.VertaModelBase.predict`
+    method to collect logs.
+
+    .. note::
+        Multithreading of calls to log() within a model's predict() method is not
+        currently supported.
+
+    Parameters
+    ----------
+    key : str
+        String value to use as data label (key) in logs, with a limit of
+        100 characters or less.
+    value : Any
+        Any `JSON serializable <https://docs.python.org/3/library/json.html#json.JSONEncoder>`__
+        value you wish to include in the logs.
+    Returns
+    -------
+    None
+
+    Raises
+    -------
+    TypeError
+        If `validate` was set to ``True`` on the active :class:`context`, and
+        `value` is not a JSON-serializable type.
+    ValueError
+        If `key` provided contains non-alphanumeric characters.  Dashes `-`
+        and underscores `_` are permitted.
+
+    Examples
+    --------
+    .. code-block:: python
+       :emphasize-lines: 15
+
+        from verta import runtime
+
+        # Sample model code:
+        class MyModel(VertaModelBase):
+            def __init__(self, artifacts):
+                pass
+
+            @verify.io
+            def predict(self, x):
+                embeddings = self.get_embeddings(x)
+                return self.nn(embeddings)
+
+            def get_embeddings(x):
+                embedding = self.embedding[x]
+                client.runtime.log("embedding", embedding)
```

Comment:
To match the `from verta import runtime` above:

```suggestion
                runtime.log("embedding", embedding)
```

Reply:
It doesn't appear that this was resolved.